tree/30_generic_methods/variable_from_augeas.cf

```diff
@@ -0,0 +1,96 @@
+#####################################################################################
+# Copyright 2020 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+#
+# @name Variable from Augeas 
+# @description Use Augeas binary to call Augtool command and get a node's label value.
+# @documentation There is two way you can use this method, either you simply provide a path to the node label as a parameter or you can specify a file associated with a lens then the regular path. 
+# The generic method will get a node's label vaule otherwise if the augeas isn't installed on the agent, it will produces an error.
+#
+# #### Two uses cases examples:
+# 
+# # Use case 1: To simply get the value of the ip address of the first line in the '/etc/hosts' file ('label' and 'value' are naming examples of 'variable_prefix' and 'variable_name')
+# variable_from_augeas("label","value","etc/hosts/1/ipaddr");
+# # Use case 2: To associate the 'Hosts' lens with the '/etc/hosts' file then get the node value
+# variable_from_augeas("label","value","Hosts","/etc/hosts","etc/hosts/1/ipaddr");
+#
+# @parameter variable_prefix The prefix of the variable name
+# @parameter variable_name   The variable to define, the full name will be variable_prefix.variable_name
+# @parameter path            The path to the file and node label
+# @parameter lens            The lens specified by the user in case he wants to load a specified lens associated with its file
+# @parameter file            The file specified by the user in case he wants to load a specified file associated with its lens
+# @parameter_constraint file "allow_empty_string" : true
+# @parameter_constraint lens "allow_empty_string" : true
+# @class_prefix variable_from_augeas
+# @class_parameter path
```

Comment:
class_parameter should be `variable_name` (as for other variable generic methods)